components/AppNav/AppNav.tsx

```diff
@@ -78,13 +78,10 @@ const NotLoggedInAuthNav = () => (
 )
 
 const AppNav: React.FC<{}> = () => {
-  const [session, setSession] = useState<GetAppQuery['session']>()
+  const [session, setSession] = useState<GetSessionQuery['session']>()
```

Comment:
Just want to confirm, by changing this line to GetSession instead of GetApp, it removes the issue of sending the same refreshToken twice to discord API?

Reply:
Changing the fetch policy from `network-only`  to `cache-first` is what fixes the issue you mentioned. When it's `cache-first`, it doesn't send a request to the GraphQL API.

But changing the query from `getApp` to `getSession`, fixes the issue of the user Discord avatar not appearing after they connect to Discord. 

In order for a query (in our case, `getSession`) to get its data from the cache, an identical query must be stored there first.

[In the curriculum page](https://github.com/garageScript/c0d3-app/blob/8076c7a4f22307a36c96fb44d02a7934ac536273/pages/curriculum.tsx#L99-L103), a `getSession` query is issued and stored in the cache. When the `AppNav` component renders, it gets the `getSession` query data that was stored from the curriculum page's `getSession` query.

Reply:
By default, the fetch policy is `cache-first` that's why the `fetchPolicy` property is removed.